app/controllers/PaypalController.scala

```diff
@@ -121,4 +149,27 @@ class PaypalController(
         Forbidden("Request isn't signed by Paypal")
     }
   }
+  case class MetadataUpdate(marketingOptIn: Boolean)
+
+  val metadataUpdateForm: Form[MetadataUpdate] = Form(
+    mapping(
+      "marketingOptIn"->boolean
+    )(MetadataUpdate.apply)(MetadataUpdate.unapply)
+  )
+
+
+  def updateMetadata(countryGroup: CountryGroup) = NoCacheAction.async { implicit request =>
```

Comment:
If you want to use a form I believe you can use `NoCacheAction(parse.form(metadataUpdateForm))` and then you don't have to do the `bindFromRequest`.